src/codec/statsd.rs

```diff
@@ -163,14 +163,14 @@ fn decode<'input>(data: &'input [u8], _ingest_ns: u64) -> Result<Value<'input>>
         Some((_, b'|')) => {
             if let Some((sample_start, b'@')) = d.next() {
                 let s = str::from_utf8(&data[sample_start + 1..])?;
-                let v: f64 = s.parse()?;
+                let v: f64 = s.trim().parse()?;
                 m.insert("sample_rate".into(), Value::from(v));
             } else {
                 return Err(ErrorKind::InvalidStatsD.into());
             }
         }
         None => (),
-        _ => return Err(ErrorKind::InvalidStatsD.into()),
+        _ => (),//return Err(ErrorKind::InvalidStatsD.into()),
```

Comment:
It could turn out problematic to silently drop invalid responses, that might make it hard to find mistakes, is there a  reason that's not visible from the snippet that needs to be commented out?

Reply:
Yeah. It is basically failing at new line character. So changed the code to ignore the new line char.